skfem/visuals/matplotlib.py

```diff
@@ -137,22 +137,14 @@ def plot_meshline(m: MeshLine, z: ndarray, **kwargs):
     else:
         ax = kwargs["ax"]
 
-    xs = []
-    ys = []
-
     color = kwargs["color"] if "color" in kwargs else 'ko-'
-    for y1, y2, s, t in zip(z[m.t[0]],
-                            z[m.t[1]],
-                            m.p[0, m.t[0]],
-                            m.p[0, m.t[1]]):
-        xs.append(s)
-        xs.append(t)
-        xs.append(None)
-        ys.append(y1)
-        ys.append(y2)
-        ys.append(None)
-
-    ax.plot(xs, ys, color)
+    ax.plot(
+        *[
+            np.vstack([a[m.t], np.full(m.t.shape[1:], None)]).flatten("F")
+            for a in [m.p[0], z]
```

Comment:
Need to check generality of this. The previous uses only first two rows of `m.t` so should it be `m.t[:2]`? Need to check whether the first two points are the termini.

Reply:
Or is this only for two-point line-elements?

Reply:
So it's invoked for `MeshLine`

https://github.com/kinnala/scikit-fem/blob/d08af2a3f4607e28023656b1b6ad025b1a530aa1/skfem/visuals/matplotlib.py#L130-L131

which is [`MeshLine1`](https://github.com/kinnala/scikit-fem/blob/d08af2a3f4607e28023656b1b6ad025b1a530aa1/skfem/mesh/mesh_line_1.py#L13)

https://github.com/kinnala/scikit-fem/blob/d08af2a3f4607e28023656b1b6ad025b1a530aa1/skfem/mesh/__init__.py#L38

and that does only cover two-point line-segments defined by their termini.

However, it remains to consider the array of ordinates `z`, which might be defined on a basis other than `ElementLineP1`.

Reply:
Right, no, it fails for
```python
element = ElementLineP2()
```
raising `ValueError: dimension mismatch`.

Reply:
&hellip;but that was only because there was a bunch of other stuff that was only going to work with P1; e.g. 

https://github.com/kinnala/scikit-fem/blob/5b788e79ff5280b56a0b787e07565e604b03fd3b/docs/examples/ex39.py#L56

https://github.com/kinnala/scikit-fem/blob/5b788e79ff5280b56a0b787e07565e604b03fd3b/docs/examples/ex39.py#L82

I've fixed all that now; the element can be changed from `ElementLineP1` or `ElementLineP2` without changing anything else.